src/ucp/wireup/wireup.c

```diff
@@ -606,6 +610,54 @@ ucp_wireup_process_reply(ucp_worker_h worker, ucp_ep_h ep,
     }
 }
 
+static UCS_F_NOINLINE void
+ucp_wireup_process_ep_check(ucp_worker_h worker, ucp_ep_h ep,
+                            const ucp_wireup_msg_t *msg,
+                            const ucp_unpacked_address_t *remote_address)
+{
+    ucs_status_t status;
+    ucp_ep_h reply_ep;
+    unsigned addr_indices[UCP_MAX_LANES];
+    ucs_status_ptr_t req;
+
+    if (ep != NULL) {
```

Comment:
do we need to check for `FAILED|CLOSED` flags?

Reply:
not needed - failed & closed EP's will be destroyed later and will be processed on next ep_check iteration

Reply:
pls add comment: if endpoint exists - do not sent a reply